dashboard/app/helpers/levels_helper.rb

```diff
@@ -107,6 +107,16 @@ def get_channel_for(level, user = nil)
     channel_token&.channel
   end
 
+  # If given a level and a user and optional script, returns whether the level
```

Comment:
Why is script optional?

Could you elaborate a bit on what we define as having "started" a level?

Reply:
I made the script optional because in the app_options method where `level_started?` is being used, we're checking for the presence of script several times. Which leads me to believe there is a case where we're getting app options without a script.

Reply:
Hmmm, not sure how best to handle this.  For querying progress, I think we need to have both (since levels can be used in multiple scripts).

(Also, there was a second question about what it means to "start" a level above.)

Reply:
I updated the comment with more information about what we mean by "started". Honestly, I was just following how the previous code defined "started", so I had to dig into when a channel and user level are created to answer this question. For channel-backed levels, a channel is created as soon as the user visits the level page, so that's when it's considered "started". For other levels the user_level is created when progress is first saved, so that's when it's considered "started".

Reply:
For the optional script, I could make it required to get `level_started` data and then wait to see if anyone reports an issue? For displaying this banner, I can't imagine a case where we'd want it outside of the context of a script.

Reply:
Thanks for the additional comment.  I think this is fine for now but may be something we revisit in the future to see if we want to make it more consistent.  (After reading your comment, I'm now kind of wondering why the logic is even different for channel-backed levels.  I'm guessing it has something to do with when we want to show this banner for template levels but it might be worth chatting with Amanda about this at some point to make sure that the behavior is what we want.)